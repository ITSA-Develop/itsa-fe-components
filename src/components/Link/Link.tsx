import { Typography } from 'antd';
import { LinkProps } from 'antd/es/typography/Link';

const { Link: AntLink } = Typography;

// TODO: double check this works as expected for proper routing
export const Link = ({ children, ...rest }: LinkProps) => {
	return <AntLink {...rest}>{children}</AntLink>;
};
