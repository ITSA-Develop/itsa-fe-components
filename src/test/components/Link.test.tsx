import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Link } from '../../components/Link/Link';

describe('Link component', () => {
    it('renders anchor with href and children', () => {
        const { container } = render(<Link href="https://example.com">Ir</Link>);
        const anchor = screen.getByRole('link', { name: 'Ir' });
        expect(anchor).toHaveAttribute('href', 'https://example.com');
        expect(container).toMatchSnapshot();
    });
});


