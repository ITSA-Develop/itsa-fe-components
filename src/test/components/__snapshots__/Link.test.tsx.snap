// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link component > renders anchor with href and children 1`] = `
<div>
  <a
    class="ant-typography css-dev-only-do-not-override-l9pxc0"
    href="https://example.com"
  >
    Ir
  </a>
</div>
`;
